examples/universal/z_bytes.cxx

```diff
@@ -0,0 +1,181 @@
+        }
+        // Corresponding encoding to be used in operations like `.put()`, `.reply()`, etc.
+        const auto encoding = Encoding("zenoh/string");
+    }
+
+    // Vec<u8>: The deserialization should be infallible
+    {
+        const std::vector<uint8_t> input = {1, 2, 3, 4};
+        const auto payload = Bytes::serialize(input);
+        const auto output = payload.deserialize<std::vector<uint8_t>>();
+        assert(input == output);
+        // Corresponding encoding to be used in operations like `.put()`, `.reply()`, etc.
+        const auto encoding = Encoding("zenoh/bytes");
+    }
+
+    // Writer & Reader
+    {
+        // serialization
+        Bytes bytes;
+        auto writer = bytes.writer();
+
+        const uint32_t i1 = 1234;
+        const std::string i2 = "test";
+        const std::vector<uint8_t> i3 = {1, 2, 3, 4};
+
+        writer.append_bounded(i1);
+        writer.append_bounded(i2);
+        writer.append_bounded(i3);
+
+        // deserialization
+        auto reader = bytes.reader();
+
+        const auto o1 = reader.read_bounded().deserialize<uint32_t>();
+        const auto o2 = reader.read_bounded().deserialize<std::string>();
+        const auto o3 = reader.read_bounded().deserialize<std::vector<uint8_t>>();
+
+        assert(i1 == o1);
+        assert(i2 == o2);
+        assert(i3 == o3);
+    }
+
+    // Iterator
+    {
+        const int32_t input[] = {1, 2, 3, 4};
+        const auto payload = Bytes::serialize_from_iter(input, input + 4);
+
+        auto idx = 0;
+        auto it = payload.iter();
+        for (auto elem = it.next(); elem.has_value(); elem = it.next()) {
+            assert(input[idx++] == elem.value().deserialize<int32_t>());
+        }
+    }
+
+    // Iterator RAW
+    {
+        const std::vector<uint8_t> input = {1, 2, 3, 4};
+        const auto payload = Bytes::serialize(input);
+
+        size_t idx = 0;
+        auto it = payload.slice_iter();
+        for (auto elem = it.next(); elem.has_value(); elem = it.next()) {
+            const auto& slice = elem.value();
+            for (size_t i = 0; i < slice.len; ++i) {
+                assert(input[idx++] == slice.data[i]);
+            }
+        }
+    }
+
+    // HashMap
+    {
+        const std::unordered_map<uint64_t, std::string> input = {{0, "abc"}, {1, "def"}};
+        const auto payload = Bytes::serialize(input);
+        const auto output = payload.deserialize<std::unordered_map<uint64_t, std::string>>();
+        assert(input == output);
+    }
+
+#ifdef ZENOH_CPP_EXMAPLE_WITH_PROTOBUF
+    // Protobuf
+    // This example is conditionally compiled depending on build system being able to find Protobuf installation
+    {
+        // (Protobuf recommendation) Verify that the version of the library that we linked against is
+        // compatible with the version of the headers we compiled against.
+        GOOGLE_PROTOBUF_VERIFY_VERSION;
+
+        // Construct PB message
+        Book input;
```

Comment:
Please use same values as https://github.com/eclipse-zenoh/zenoh-python/blob/90e74618e0aa16e6677c26e8a0c6ef44dfe8c7b2/examples/z_bytes.py#L72 and https://github.com/eclipse-zenoh/zenoh/blob/8b2566e543a17e368708874f73789065ca4c5c5c/examples/examples/z_bytes.rs#L154